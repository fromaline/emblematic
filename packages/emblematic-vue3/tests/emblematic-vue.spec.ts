import { afterEach, describe, expect, it, vi } from 'vitest'
import { cleanup, fireEvent, render } from '@testing-library/vue'
import { Algorand, Arbitrum, Avalanche, Base, Bitcoin, BnbChain, BnbGreenfield, Celo, Cronos, Crust, Ethereum, Fantom, Gnosis, Hedera, Integritee, Iotex, Klaytn, Moonbeam, Opbnb, Optimism, Polkadex, Polkadot, Polygon, Scroll, Sirius, Solana, Starknet, Telos, Tron, Wanchain, Zeitgeist } from '../src/emblematic-vue3'

describe('using Emblematic icon components', () => {
  afterEach(() => cleanup())

  const icons = [Algorand, Arbitrum, Avalanche, Base, Bitcoin, BnbChain, BnbGreenfield, Celo, Cronos, Crust, Ethereum, Fantom, Gnosis, Hedera, Integritee, Iotex, Klaytn, Moonbeam, Opbnb, Optimism, Polkadex, Polkadot, Polygon, Scroll, Sirius, Solana, Starknet, Telos, Tron, Wanchain, Zeitgeist]

  it('should render a component', () => {
    icons.forEach((icon) => {
      const { container } = render(icon)
      expect(container).toMatchSnapshot()
    })
  })

  it('should adjust the size', () => {
    const { container } = render(Ethereum, {
      props: {
        size: 48,
      },
    })

    expect(container).toMatchSnapshot()
  })

  it('should add a class to the element', () => {
    const testClass = 'my-icon'

    const { container } = render(Ethereum, {
      props: {
        class: testClass,
      },
    })

    expect(container).toMatchSnapshot()

    const el = document.querySelector(`.${testClass}`)

    expect(el).toBeInTheDocument()
    expect(el).toMatchSnapshot()
    expect(el).toHaveClass(testClass)
    expect(el).toHaveClass('emblematic')
    expect(el).toHaveClass('emblematic-ethereum-icon')
  })

  it('should add a style attribute to the element', () => {
    const { container } = render(Ethereum, {
      props: {
        style: 'position: absolute;',
      },
    })

    expect(container).toMatchSnapshot()

    const el = document.querySelector('.emblematic')

    expect(el).toHaveStyle({ position: 'absolute' })
  })

  it('should call the onClick event', async () => {
    const onClick = vi.fn()
    render(Ethereum, {
      attrs: {
        onClick,
      },
    })

    const el = document.querySelector('.emblematic')

    if (el) {
      await fireEvent.click(el)

      expect(onClick).toHaveBeenCalled()
    }
  })

  it('should pass children to the icon slot', () => {
    const testText = 'Temp Text'
    const template = {
      name: 'Temp',
      template: `<text>${testText}</text>`,
    }

    const { getByText, container } = render(Ethereum, {
      slots: {
        default: template,
      },
    })

    const textElement = getByText(testText)

    expect(textElement).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
