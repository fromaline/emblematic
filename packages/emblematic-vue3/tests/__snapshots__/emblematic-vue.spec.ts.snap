// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`using Emblematic icon components > should add a class to the element 1`] = `
<div>
  <svg
    class="emblematic emblematic-ethereum-icon my-icon"
    height="32"
    viewBox="0 0 32 32"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      type="text"
      values="Ethereum"
    >
      Ethereum
    </title>
    <path
      d="M16 0 6.2 16.3l9.8-4.5V0Z"
      fill="#8891B1"
    />
    <path
      d="m16 11.8-9.8 4.5 9.8 5.8V11.8Zm9.8 4.5L16 0v11.8l9.8 4.5Z"
      fill="#63688E"
    />
    <path
      d="m16 22.1 9.8-5.8-9.8-4.5v10.3Z"
      fill="#464A75"
    />
    <path
      d="M6.2 18.2 16 32v-8l-9.8-5.8Z"
      fill="#8891B1"
    />
    <path
      d="M16 24v8l9.8-13.8L16 24Z"
      fill="#63688E"
    />
  </svg>
</div>
`;

exports[`using Emblematic icon components > should add a class to the element 2`] = `
<svg
  class="emblematic emblematic-ethereum-icon my-icon"
  height="32"
  viewBox="0 0 32 32"
  width="32"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    type="text"
    values="Ethereum"
  >
    Ethereum
  </title>
  <path
    d="M16 0 6.2 16.3l9.8-4.5V0Z"
    fill="#8891B1"
  />
  <path
    d="m16 11.8-9.8 4.5 9.8 5.8V11.8Zm9.8 4.5L16 0v11.8l9.8 4.5Z"
    fill="#63688E"
  />
  <path
    d="m16 22.1 9.8-5.8-9.8-4.5v10.3Z"
    fill="#464A75"
  />
  <path
    d="M6.2 18.2 16 32v-8l-9.8-5.8Z"
    fill="#8891B1"
  />
  <path
    d="M16 24v8l9.8-13.8L16 24Z"
    fill="#63688E"
  />
</svg>
`;

exports[`using Emblematic icon components > should add a style attribute to the element 1`] = `
<div>
  <svg
    class="emblematic emblematic-ethereum-icon"
    height="32"
    style="position: absolute;"
    viewBox="0 0 32 32"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      type="text"
      values="Ethereum"
    >
      Ethereum
    </title>
    <path
      d="M16 0 6.2 16.3l9.8-4.5V0Z"
      fill="#8891B1"
    />
    <path
      d="m16 11.8-9.8 4.5 9.8 5.8V11.8Zm9.8 4.5L16 0v11.8l9.8 4.5Z"
      fill="#63688E"
    />
    <path
      d="m16 22.1 9.8-5.8-9.8-4.5v10.3Z"
      fill="#464A75"
    />
    <path
      d="M6.2 18.2 16 32v-8l-9.8-5.8Z"
      fill="#8891B1"
    />
    <path
      d="M16 24v8l9.8-13.8L16 24Z"
      fill="#63688E"
    />
  </svg>
</div>
`;

exports[`using Emblematic icon components > should adjust the size 1`] = `
<div>
  <svg
    class="emblematic emblematic-ethereum-icon"
    height="48"
    size="48"
    viewBox="0 0 32 32"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      type="text"
      values="Ethereum"
    >
      Ethereum
    </title>
    <path
      d="M16 0 6.2 16.3l9.8-4.5V0Z"
      fill="#8891B1"
    />
    <path
      d="m16 11.8-9.8 4.5 9.8 5.8V11.8Zm9.8 4.5L16 0v11.8l9.8 4.5Z"
      fill="#63688E"
    />
    <path
      d="m16 22.1 9.8-5.8-9.8-4.5v10.3Z"
      fill="#464A75"
    />
    <path
      d="M6.2 18.2 16 32v-8l-9.8-5.8Z"
      fill="#8891B1"
    />
    <path
      d="M16 24v8l9.8-13.8L16 24Z"
      fill="#63688E"
    />
  </svg>
</div>
`;

exports[`using Emblematic icon components > should pass children to the icon slot 1`] = `
<div>
  <svg
    class="emblematic emblematic-ethereum-icon"
    height="32"
    viewBox="0 0 32 32"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      type="text"
      values="Ethereum"
    >
      Ethereum
    </title>
    <path
      d="M16 0 6.2 16.3l9.8-4.5V0Z"
      fill="#8891B1"
    />
    <path
      d="m16 11.8-9.8 4.5 9.8 5.8V11.8Zm9.8 4.5L16 0v11.8l9.8 4.5Z"
      fill="#63688E"
    />
    <path
      d="m16 22.1 9.8-5.8-9.8-4.5v10.3Z"
      fill="#464A75"
    />
    <path
      d="M6.2 18.2 16 32v-8l-9.8-5.8Z"
      fill="#8891B1"
    />
    <path
      d="M16 24v8l9.8-13.8L16 24Z"
      fill="#63688E"
    />
    
    <text>
      Temp Text
    </text>
    
  </svg>
</div>
`;

exports[`using Emblematic icon components > should render a component 1`] = `
<div>
  <svg
    class="emblematic emblematic-ethereum-icon"
    height="32"
    viewBox="0 0 32 32"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      type="text"
      values="Ethereum"
    >
      Ethereum
    </title>
    <path
      d="M16 0 6.2 16.3l9.8-4.5V0Z"
      fill="#8891B1"
    />
    <path
      d="m16 11.8-9.8 4.5 9.8 5.8V11.8Zm9.8 4.5L16 0v11.8l9.8 4.5Z"
      fill="#63688E"
    />
    <path
      d="m16 22.1 9.8-5.8-9.8-4.5v10.3Z"
      fill="#464A75"
    />
    <path
      d="M6.2 18.2 16 32v-8l-9.8-5.8Z"
      fill="#8891B1"
    />
    <path
      d="M16 24v8l9.8-13.8L16 24Z"
      fill="#63688E"
    />
  </svg>
</div>
`;
